src/main/java/org/example/jakarta_labb/resource/MovieResource.java

```diff
@@ -0,0 +1,53 @@
+package org.example.jakarta_labb.resource;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.MediaType;
+import org.example.jakarta_labb.entity.Movie;
+import org.example.jakarta_labb.repository.MovieRepository;
+
+import java.util.List;
+import java.util.UUID;
+
+@Path("/movies")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class MovieResource {
+
+    @Inject
```

Comment:
Might be good to change the @Inject annotation to a constructor injection instead. Sonar mentioned this as well as [Issue](https://sonarcloud.io/project/issues?resolved=false&pullRequest=26&id=Lucia560_Jakarta_labb&open=AY3dLnPWY7d5sJwlFNgd&tab=code). 

> By using constructor injection, the dependencies are explicit and must be passed during an object’s construction. This avoids the possibility of instantiating an object in an invalid state and makes types more testable. Fields can be declared final, which makes the code easier to understand, as dependencies don’t change after instantiation.

Read more [here](https://sonarcloud.io/project/issues?resolved=false&pullRequest=26&id=Lucia560_Jakarta_labb&open=AY3dLnPWY7d5sJwlFNgd&tab=how_to_fix) on how to fix this issue.

Reply:
i thought so too, but wanted to wait for confirmation! Will get right on it!